src/routes/user.ts

```diff
@@ -0,0 +1,88 @@
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import { Type, Static } from '@sinclair/typebox';
+import { knexController } from '../database/database';
+
+const userAddedResponse = Type.Object({
+  status: Type.String(),
+});
+
+export type UserAddedResponse = Static<typeof userAddedResponse>;
+
+const schema = {
+  description: 'Add new user or return error if add is not successful',
+  summary: 'Add user',
+  tags: ['User'],
+  body: {
+    type: 'object',
+    required: ['email', 'forename', 'surname', 'password'],
+    properties: {
+      email: {
+        type: 'string',
+      },
+      forename: {
+        type: 'string',
+      },
+      surname: {
+        type: 'string',
+      },
+      password: {
+        type: 'string',
+      },
+    },
+  },
+  response: {
+    201: {
+      description: 'User added',
+      type: 'null',
+    },
+    409: {
+      description: 'Email already in use',
+      type: 'null',
+    },
+    500: {
+      description: 'Server error',
+      type: 'null',
+    },
```

Comment:
Käytetään standardeja error schemoja. Jos käyt vilkaisemassa server.ts, niin riveillä 30-37 määritellään schema, jota käytetään kaikissa erroreissa.

Tällöin esim 409 määrittely tässä olisi

```
409: {
      $ref: 'error',
    },
```

Reply:
Uus tuttavuus mulle